pkg/solana/txm/txm.go

```diff
@@ -926,6 +992,34 @@ func (txm *Txm) InflightTxs() int {
 	return len(txm.txs.ListAllSigs())
 }
 
+// rebroadcastWithGivenBlockhash attempts to rebroadcast a pending tx with a new blockhash.
+// Removes all signatures associated with the prior tx, cancels prior ctx, updates compute unit price and sets given blockhash for rebroadcasting.
+// Calls sendWithRetry directly to avoid enqueuing the transaction. It logs the error when rebroadcast fails and returns the new signature when successful.
+func (txm *Txm) rebroadcastWithGivenBlockhash(ctx context.Context, pTx pendingTx, blockhash solana.Hash, lastValidBlockHeight uint64) (solana.Signature, error) {
+	// Removes all signatures associated to prior tx and cancels context.
```

Comment:
nit: Just say "Remove the previous tx from state".
Because it does that, in addition to signatures.